src/components/Input.react.js

```diff
@@ -41,6 +41,20 @@ export default class Input extends Component {
                     if (fireEvent) {
                         fireEvent({event: 'blur'});
                     }
+                    if (setProps) {
```

Comment:
A small improvement here could be that if `setProps` is set, the component doesn't really need to keep state itself, so `this.setState` should only be called if `setProps` is empty.